src/matches.rs

```diff
@@ -103,9 +103,15 @@ pub(crate) fn rewrite_match(
     let inner_attrs_str = if inner_attrs.is_empty() {
         String::new()
     } else {
-        inner_attrs
-            .rewrite(context, shape)
-            .map(|s| format!("{}{}\n", nested_indent_str, s))?
+        if context.config.version() == Version::One {
+            inner_attrs
+                .rewrite(context, shape)
+                .map(|s| format!("{}{}\n", nested_indent_str, s))?
+        } else {
+            inner_attrs
+                .rewrite(context, shape.block_indent(context.config.tab_spaces()))
+                .map(|s| format!("{}{}\n", nested_indent_str, s))?
+        }
```

Comment:
I think it'll be simpler and avoid duplication if we version gate the `shape` calculation instead of the attribute rewrite. (code might need formatting)

```suggestion
        let shape = if context.config.version() == Version::One {
            shape
        } else {
            shape.block_indent(context.config.tab_spaces())
        };
        inner_attrs
            .rewrite(context, shape)
            .map(|s| format!("{}{}\n", nested_indent_str, s))?
```